### ISSUE TITLE
origin field meaning should be more nuanced.

### ISSUE THREAD

**Issue by kardianos** [kardianos/vendor-spec#36]
@rsc @nathany @davecheney @freeformz
I'm finishing bringing kardianos/govendor in line with the recent vendor-spec changes. I would like to change the "origin" field definition slightly.
- empty or missing origin indicates either: the package was copied from the "path" or it doesn't matter where it comes from.
- origin field is the same as the path indicates: when updating the package the source should be the same as the path (no prompting required).
- origin and path fields are different indicates: when updating the package the source should come from the origin field.

In short, when updating and multiple sources exist (either from other vendor folders or GOPATH or remote), an empty origin indicates no preference (tool or developer decides). When the origin is non-empty (regardless whether it is the same or different then path) it indicates there is a preference where the package should come from.

This is different than what is currently specified where missing or empty origin field is equivalent to the origin field being equal to the path field.

I'd appreciate your feedback. Thanks.

**Comment by rsc** [comment]
The first bullet seems like a category error to me. The vendor.json file is not meant to record where a future copy should come from. It records where the current copy came from. It came from somewhere (or else why does vendor.json even mention it?). Basic operations like showing diffs for local changes or updating to a new copy require being able to identify where the source came from.

The overwhelmingly common case is that the path and the origin are the same. The file should not be required to record both in this case. People read and debug these files. The common case should be simple, as short as possible.

I don't really understand what "it doesn't matter where it comes from" could mean, but ambiguity here seems like a mistake. This is not a file recording preferences. It records history.

**Comment by kardianos** [comment]
@rsc I do agree with you.

I was writing out a reply to explain my problem when I realized I don't actually have a problem. When adding it initially if origin should be different, they will choose it at that time. Thank you for your reply and letting me talk it out. I appreciate it :).
